### PR TITLE
OCPQE-25882: inject correct version of oc cli in chain upgrade ci

### PR DESCRIPTION
**Pull request by jianlinliu**
1. Remove openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-cpou-upgrade-from-4.10.yaml, we do not run stable cpou upgrade ci.
2. Introduce `openshift-upgrade-qe-sanity-arch-migration-upi` to replace `openshift-upgrade-qe-test-arch-migration-upi`, because we do not run e2e test suite in stable ci.
3. In favor of https://github.com/openshift/release/pull/57024, reference `openshift-extended-upgrade-pre-custom-cli` in chain upgrade step, set `release:custom` image in chian upgrade ci yaml to ensure inject final-1 version of cli, so that `openshift-extended-upgrade-pre` always works.

### CODE REVIEW

**Comment by jianlinliu** [comment]
cc @jiajliu @shellyyang1989 @kuiwang02 @JianLi-RH to review.

**Comment by jianlinliu** [comment]
@shellyyang1989 I did not which job hit OCPQE-25882, can you help trigger a rehearse to confirm if it is fixed ?

**Comment by shellyyang1989** [comment]
sure Jianlin. Thanks for the fix!

/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-4.17-upgrade-from-stable-4.3-aws-ipi-f28

**Comment by kuiwang02** [comment]
I am ok for change

File: ci-operator/step-registry/openshift/upgrade/qe/test/custom-cli/openshift-upgrade-qe-test-custom-cli-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+chain:
+  as: openshift-upgrade-qe-test-custom-cli
+  steps:
+  - ref: idp-htpasswd
+  - ref: cucushift-pre
+  - ref: openshift-extended-upgrade-pre-custom-cli
+  - ref: cucushift-upgrade-prepare
+  - ref: cucushift-upgrade-prehealthcheck
+  - ref: cucushift-upgrade-toimage
+  - ref: cucushift-upgrade-healthcheck
```
Comment (jiajliu) [inline ci-operator/step-registry/openshift/upgrade/qe/test/custom-cli/openshift-upgrade-qe-test-custom-cli-chain.yaml:10]: I see there is profile with none basecap in chain upgrade jobs, shall we keep step `cucushift-upgrade-check-capability` in the chain?

Comment (jianlinliu) [reply ci-operator/step-registry/openshift/upgrade/qe/test/custom-cli/openshift-upgrade-qe-test-custom-cli-chain.yaml:10]: Okay, later I will add it back.

**Comment by jianlinliu** [comment]
> sure Jianlin. Thanks for the fix!
> 
> /pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-4.17-upgrade-from-stable-4.3-aws-ipi-f28

`openshift-extended-upgrade-pre-custom-cli` is completed, can you help check if the issue is fixed now ?

**Comment by jianlinliu** [comment]
Explore the job history:

Before this fix, we can see the following error in the job [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-4.17-upgrade-from-stable-4.3-aws-ipi-f28/1836398666059878400/artifacts/aws-ipi-f28/openshift-extended-upgrade-pre/build-log.txt):

```
Error: unknown flag: --port
```

After this fix, we can not see the error in the pj-rehearse job [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/57077/rehearse-57077-periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-4.17-upgrade-from-stable-4.3-aws-ipi-f28/1838510393517936640/artifacts/aws-ipi-f28/openshift-extended-upgrade-pre-custom-cli/build-log.txt)

/pj-rehearse ack

File: ci-operator/step-registry/openshift/upgrade/qe/test/custom-cli/openshift-upgrade-qe-test-custom-cli-chain.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/openshift/upgrade/qe/test/custom-cli/openshift-upgrade-qe-test-custom-cli-chain.yaml:10]: Done.

**Comment by jiajliu** [comment]
lgtm, leave it to yang to confirm https://github.com/openshift/release/pull/57077#issuecomment-2373085302

**Comment by shellyyang1989** [comment]
The e2e test doesn't complain the old oc version any more. Thanks for the fix.

/lgtm

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Comment by liangxia** [comment]
@jianlinliu  There are conflicts.

**Comment by jianlinliu** [comment]
@liangxia conflicts is resolved, pls help review again, thanks.

**Pull request merged**